images/win/scripts/Installers/Install-BizTalkBuildComponent.ps1

```diff
@@ -0,0 +1,89 @@
+################################################################################
+##  File:  Install-BizTalkBuildComponent.ps1
+##  Desc:  Install BizTalk Project Build Component
+################################################################################
+
+function Install-Msi
+{
+    <#
+    .SYNOPSIS
+        A helper function to install executables.
+
+    .DESCRIPTION
+        install .exe or .msi binaries from specified Path.
+
+    .PARAMETER MsiPath
+        Msi or exe path. Required parameter.
+
+    .PARAMETER LogPath
+        The log file path where installation will write log to. Required parameter.
+
+    .EXAMPLE
+        Install-Msi -MsiPath "c:\temp\abc.msi" -LogPath "c:\abc.log"
+    #>
+
+    Param
+    (
+        [Parameter(Mandatory)]
+        [String] $MsiPath,
+        [Parameter(Mandatory)]
+        [String] $LogPath
+    )
+
+    try
+    {
+        $filePath = "msiexec.exe"
+
+        Write-Host "Starting Install $MsiPath..."
+        $ArgumentList = ('/i', $MsiPath, '/QN', '/norestart', "/l*v",$LogPath)
+        $process = Start-Process -FilePath $filePath -ArgumentList $ArgumentList -Wait -PassThru -Verb runAs
+
+        $exitCode = $process.ExitCode
+        if ($exitCode -eq 0 -or $exitCode -eq 3010)
+        {
+            Write-Host "Installation for $MsiPath is successful."
+        }
+        else
+        {
+            Write-Host "Non zero exit code returned by $MsiPath installation process: $exitCode"
+            Get-Content  $LogPath | Write-Host
+            exit $exitCode
+        }
+    }
+    catch
+    {
+        Write-Host "Failed to install $MsiPath : $($_.Exception.Message)"
+        exit 1
+    }
+}
+
+$bizTalkBuildComponentUri = "https://aka.ms/BuildComponentSetup.EN"
+
+# Download
+try {
+    Write-Host "BizTalk Project Build Component download..."
+    $setupZipFile = Start-DownloadWithRetry -Url $bizTalkBuildComponentUri -Name "BuildComponentSetup.EN.zip"
+}
+catch {
+    Write-Error "[!] Failed to download $bizTalkBuildComponentUri"
+    exit 1
+}
+
+# Unzip
+$setupPath = "C:\BizTalkBuildComponent"
+if (-not (Test-Path -Path $setupPath)) {
+    $null = New-Item -Path $setupPath -ItemType Directory -Force
+}
+
+Write-Host "Unzip $setupZipFile to $setupPath..."
+Extract-7Zip -Path $setupZipFile -DestinationPath $setupPath
+Remove-Item $setupZipFile
+
+# Install
+Install-Msi -MsiPath "$setupPath\Bootstrap.msi" -LogPath "$setupPath\bootstrap.log"
```

Comment:
Are those MSI files are available by direct links?
`Install-Binary` helper can download and install that can significantly simplify script: https://github.com/actions/virtual-environments/blob/main/images/win/scripts/ImageHelpers/InstallHelpers.ps1#L1

Reply:
No, our package contains MSI and a bunch of other files, so we make it a Zip, we have to unzip it and then install from local disk. I checked Install-Binary, which won't help in this situation, unless you want me to add this Install-Msi to InstallHelpers.ps1.

Reply:
@Elvis-Shi, What do you think about to rework the `Invoke-Binary` function and use ParameterSetName?